openupgradelib/openupgrade.py

```diff
@@ -3261,17 +3261,43 @@ def chunked(records, single=True):
     """Memory and performance friendly method to iterate over a potentially
     large number of records. Yields either a whole chunk or a single record
     at the time. Don't nest calls to this method."""
-    if version_info[0] > 10:
-        invalidate = records.env.cache.invalidate
-    elif version_info[0] > 7:
-        invalidate = records.env.invalidate_all
+    # Warning on possibly untested versions where chunked might not work as expected
+    if version_info[0] > 17:  # unreleased version at this time
+        logger.warning(
+            f"chunked method might not work as expected in Odoo {version_info[0]}. "
+            "Please report any issue you may find and consider bumping this warning "
+            "to the next version otherwise."
+        )
+    # Before each iteration we must flush everything and clear the cache.
+    # This differs depending on the Odoo version.
+    if version_info[0] >= 16:
+
+        def reset():
+            records.env.invalidate_all(flush=True)
+
+    elif version_info[0] >= 13:
+
+        def reset():
+            records.env["base"].flush()
+            records.env.cache.invalidate()
+
+    elif version_info[0] >= 11:
+
+        def reset():
+            records.env.cache.invalidate()
+
+    elif version_info[0] >= 8:
+
+        def reset():
```

Comment:
A version dependent shorthand method to invalidate the cache might proof to be useful outside of `chunked`. Would you consider moving it to `openupgrade_tools.py`? It might also be an aesthetic improvement, depending on your taste of course.

Reply:
Sure ;) Check out the latests changes.